jax/numpy/lax_numpy.py

```diff
@@ -2580,6 +2580,26 @@ def diag(v, k=0):
   else:
     raise ValueError("diag input must be 1d or 2d")
 
+@_wraps(np.diagflat)
+def diagflat(v, k=0):
+  if np.isscalar(v):
```

Comment:
I think this code will likely fail the compilation test, because it calls a numpy function on a jax object.

If you instead use ``ndim(v) == 0``, I expect it would work.

Reply:
Hmm.. but that may not work either. Numpy is not very consistent here:
```python
>>> np.diagflat(4, k=1)
array([[0, 4],
       [0, 0]])
>>> np.diagflat(np.float64(4), k=1)
0.0
```
I don't see any good reason why these should be different. It may be better to deliberately diverge from numpy here, document the differences, and adjust the tests appropriately.

Reply:
yeah I observed it too. Unless we explicitly make it scalar using numpy. Numpy internally promotes it and makes it array for which ```np.isscalar``` evaluates to ```False``` and ```np.diagflat``` gives expected answer. What do you suggest we do?

Reply:
I think the difference comes from the fact that ``np.diagflat`` wraps its output in ``__array_wraps__``, which is defined for numpy scalars and not defined for python scalars, whereas jax does not distinguish between python and numpy scalars.

I would say the best approach would be to always ``ravel()`` input here, document the divergence from numpy using `lax_description`, and adjust the tests to call `np.asarray()` on inputs to `np_fun` with a comment explaining why that must be done.

Reply:
I filed a numpy bug about the inconsistency; curious to see what the team says https://github.com/numpy/numpy/issues/16477

Reply:
Sorry for bugging. Should we remove all tests which uses ```scalar``` elements?

Thanks

Reply:
No, I think we want jax to return a matrix even for scalar inputs; in other words `jnp.diagflat(x)` should match `np.diagflat(np.atleast_1d(x))`. We can test this by defining the test this way:
```python
np_fun = lambda arg: np.diagflat(np.atleast_1d(arg), k)
```
and then keeping all the scalar inputs.

Then add a comment in the test explaining why we're calling ``atleast_1d``, and a ``lax_description`` in the function definition to document the difference from numpy's behavior.

Reply:
Hi I have added the above mentioned changes. I am not sure if the lax_description I wrote captures the issue. If anything needs to be changed please let me know.

Thanks